data/templates/nginx/server.tpl.conf

```diff
@@ -20,41 +20,39 @@ server {
 }
 
 server {
-    # Disabling http2 for now as it's causing weird issues with curl
-    #listen 443 ssl http2;
-    #listen [::]:443 ssl http2;
-    listen 443 ssl;
-    listen [::]:443 ssl;
+    # curl is fixed
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+    #listen 443 ssl;
+    #listen [::]:443 ssl;
     server_name {{ domain }};
 
     ssl_certificate /etc/yunohost/certs/{{ domain }}/crt.pem;
     ssl_certificate_key /etc/yunohost/certs/{{ domain }}/key.pem;
-    ssl_session_timeout 5m;
-    ssl_session_cache shared:SSL:50m;
+    ssl_session_timeout 10m;
+    ssl_session_cache shared:SSL:10m;
+    ssl_session_tickets off;
+    
+    server_tokens off;
 
     # As suggested by Mozilla : https://wiki.mozilla.org/Security/Server_Side_TLS and https://en.wikipedia.org/wiki/Curve25519
     # (this doesn't work on jessie though ...?)
     # ssl_ecdh_curve secp521r1:secp384r1:prime256v1;
 
     # As suggested by https://cipherli.st/
-    ssl_ecdh_curve secp384r1;
+    #ssl_ecdh_curve secp384r1;
 
     ssl_prefer_server_ciphers on;
 
-    # Ciphers with intermediate compatibility
-    # https://mozilla.github.io/server-side-tls/ssl-config-generator/?server=nginx-1.6.2&openssl=1.0.1t&hsts=yes&profile=intermediate
-    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
-    ssl_ciphers 'ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS';
-
     # Ciphers with modern compatibility
     # https://mozilla.github.io/server-side-tls/ssl-config-generator/?server=nginx-1.6.2&openssl=1.0.1t&hsts=yes&profile=modern
     # Uncomment the following to use modern ciphers, but remove compatibility with some old clients (android < 5.0, Internet Explorer < 10, ...)
-    #ssl_protocols TLSv1.2;
-    #ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256';
+    ssl_protocols TLSv1.2;
+    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
```

Comment:
Au départ, en fait les cyppher sont celles de Mozilla qui est un des spécialistes en sécurité pour le web (plusieurs autres bien-sûr les accompagnent dans leurs choix). Pourrait-tu donner une source ?

Reply:
cela viens d'une expertise qui vient des sources : 
https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html
https://github.com/ssllabs/research/wiki/SSL-and-TLS-Deployment-Best-Practices
https://cipherli.st/

Reply:
I think that all cipher with AES128 should be removed.

Reply:
> Mozilla qui est un des spécialistes en sécurité pour le web

When I see some issue like that : https://bugzilla.mozilla.org/show_bug.cgi?id=672600#c65

We can't be really sure that Mozilla do everything to improve the web security.

Reply:
Effectivement je pense aussi que pour moi on ne dois pas ce reposer que sur Mozilla c'est faire preuve de trop de facilité et voir que en somme d'autres apporté des solution plus efficace en sécurité que mozilla.